plugin/pkg/managedseed/validator/admission.go

```diff
@@ -239,6 +241,45 @@ func (v *ManagedSeed) Admit(ctx context.Context, a admission.Attributes, o admis
 		return apierrors.NewInvalid(gk, managedSeed.Name, allErrs)
 	}
 
+	// If the managed seed's HA configuration is changed from multi-zonal to non-multi-zonal. Only allow it if there are no multi-zonal shoots provisioned on this managed seed.
+	if a.GetOperation() == admission.Update {
+		oldManagedSeed, ok := a.GetOldObject().(*seedmanagement.ManagedSeed)
+		if !ok {
+			return apierrors.NewInternalError(errors.New("could not covert old resource into ManagedSeed object"))
+		}
+		return v.validateManagedSeedHAConfigUpdate(oldManagedSeed, managedSeed)
+	}
+
+	return nil
+}
+
+func (v *ManagedSeed) validateManagedSeedHAConfigUpdate(oldManagedSeed, newManagedSeed *seedmanagement.ManagedSeed) error {
+	seedName := newManagedSeed.Name
+	oldManagedSeedMultiZonal, err := seedmanagementhelper.IsMultiZonalManagedSeed(oldManagedSeed)
+	if err != nil {
+		return apierrors.NewInternalError(fmt.Errorf("failed to check if old managedseed: %s is multi-zonal: %w", oldManagedSeed.Name, err))
+	}
+
+	newManagedSeedMultiZonal, err := seedmanagementhelper.IsMultiZonalManagedSeed(newManagedSeed)
+	if err != nil {
+		return apierrors.NewInternalError(fmt.Errorf("failed to check if new managedseed: %s is multi-zonal: %w", newManagedSeed.Name, err))
+	}
+
+	if oldManagedSeedMultiZonal && !newManagedSeedMultiZonal {
+		//check if there are any multi-zonal shootList which have their control planes already provisioned in this managed seed.
+		multiZonalShoots, err := admissionutils.GetShoots(v.shootLister, func(shoot *gardencore.Shoot) bool {
+			return shoot.Spec.SeedName != nil &&
+				*shoot.Spec.SeedName == seedName &&
+				gardencorehelper.IsMultiZonalShootControlPlane(shoot)
+		})
+		if err != nil {
+			return err
+		}
+
+		if len(multiZonalShoots) > 0 {
+			return apierrors.NewForbidden(gardencore.Resource("ManagedSeed"), seedName, fmt.Errorf("managedseed %s cannot be changed from mulit-zonal to non-multi-zonal as there are %d shoots which are multi-zonal on this managedseed", seedName, len(multiZonalShoots)))
```

Comment:
```suggestion
			return apierrors.NewForbidden(gardencore.Resource("ManagedSeed"), seedName, fmt.Errorf("managedseed %s cannot be changed from multi-zonal to single-zonal as there are %d multi-zonal shoots on this managedseed", seedName, len(multiZonalShoots)))
```

Reply:
It is deliberately kept as non-multi-zonal as that covers both the cases for non-HA and single-zonal shoot control planes.